### PR TITLE
Merge digital twin preview code into preview branch

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Keeping commit history from the old digital twin repo as well

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -266,6 +268,8 @@ try {
         BuildPackage provisioning\transport\http\src "Provisioning Transport for HTTP"
         BuildPackage provisioning\transport\mqtt\src "Provisioning Transport for MQTT"
         BuildPackage provisioning\service\src "Provisioning Service Client SDK"
+		BuildPackage digitaltwin\device\src "Digital Twin Device Client SDK"
```
Comment (abhipsaMisra) [inline build.ps1]: nit (unimportant right now): extra space

File: digitaltwin/device/src/Microsoft.Azure.Devices.DigitalTwin.Client.csproj [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <!-- delay sign the assembly for Release build -->
```
Comment (abhipsaMisra) [inline digitaltwin/device/src/Microsoft.Azure.Devices.DigitalTwin.Client.csproj]: Why is this commented out?

File: digitaltwin/service/src/Microsoft.Azure.Devices.DigitalTwin.Service.csproj [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <!-- delay sign the assembly for Release build -->
```
Comment (abhipsaMisra) [inline digitaltwin/service/src/Microsoft.Azure.Devices.DigitalTwin.Service.csproj]: Same as above - Why is this commented out?

File: iothub/service/src/Microsoft.Azure.Devices.csproj [hunk 4]
```diff
@@ -158,4 +158,10 @@
     <PackageReference Condition=" '$(Configuration)' == 'Debug' " Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.1" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
```
Comment (abhipsaMisra) [inline iothub/service/src/Microsoft.Azure.Devices.csproj]: Why do we need AutoRest dependency on our existing service client?

File: digitaltwin/service/readme.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Azure IoT Digital Twins Service SDK
```
Comment (drwill) [inline digitaltwin/service/readme.md]: Is this readme.md different than the earlier one? It looks more or less the same, at first glance.

File: digitaltwin/service/readme.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+## API reference
+
+To be written soon
+
+## Installation
+
+Install the Azure Digital Twin Service Client library for .NET with NuGet:
+
+```
+dotnet add package Azure.IoT.DigitalTwin.Service
```
Comment (drwill) [inline digitaltwin/service/readme.md]: Microsoft.Azure.*?

File: digitaltwin/service/src/AuthorizationDelegatingHandler.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.DigitalTwin.Service
+{
+    class AuthorizationDelegatingHandler : DelegatingHandler
```
Comment (drwill) [inline digitaltwin/service/src/AuthorizationDelegatingHandler.cs]: internal?

File: digitaltwin/stylecop.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+      "usingDirectivesPlacement": "outsideNamespace",
+      "blankLinesBetweenUsingGroups": "require"
+    },
+    "layoutRules": {
+      "newlineAtEndOfFile": "require"
+    },
+    "documentationRules": {
+      "companyName": "Microsoft",
+      "xmlHeader": false,
+      "copyrightText": "Copyright (c) {companyName}. All rights reserved.\nLicensed under the MIT license. See LICENSE file in the project root for full license information.",
```
Comment (drwill) [inline digitaltwin/stylecop.json]: {companyName}?

File: build.ps1 (hunk 1 above)
Comment (drwill) [inline build.ps1]: This appears to be tabbed in one too many times

File: digitaltwin/device/src/Microsoft.Azure.Devices.DigitalTwin.Client.csproj [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  <!-- <PropertyGroup Condition=" '$(Configuration)' == 'Release' AND '$(AZURE_IOT_DONOTSIGN.ToUpper())' != 'TRUE' ">
```
Comment (drwill) [inline digitaltwin/device/src/Microsoft.Azure.Devices.DigitalTwin.Client.csproj]: Should this be uncommented?

File: iothub/service/src/Microsoft.Azure.Devices.csproj (hunk 4 above)
Comment (timtay-microsoft) [reply iothub/service/src/Microsoft.Azure.Devices.csproj]: We don't need it, I'll remove it

Comment (timtay-microsoft) [reply iothub/service/src/Microsoft.Azure.Devices.csproj]: I've removed it now, good catch

File: digitaltwin/service/src/Microsoft.Azure.Devices.DigitalTwin.Service.csproj (hunk 3 above)
Comment (timtay-microsoft) [reply digitaltwin/service/src/Microsoft.Azure.Devices.DigitalTwin.Service.csproj]: I've uncommented it now, good catch

File: digitaltwin/device/src/Microsoft.Azure.Devices.DigitalTwin.Client.csproj (hunk 2 above)
Comment (timtay-microsoft) [reply digitaltwin/device/src/Microsoft.Azure.Devices.DigitalTwin.Client.csproj]: I've uncommented it now, good catch

File: build.ps1 (hunk 1 above)
Comment (timtay-microsoft) [reply build.ps1]: Sure, I can fix this

**Comment by abhipsaMisra** [comment]
Closing this PR as this is resolved in #1162

**Pull request closed without merging**